### ISSUE TITLE
Should ldap_user be rewritten as an AuthenticationProvider?

### ISSUE THREAD

**Issue by queenvictoria** [Laudanum/ldap#7]
Authentication poviders:

* CookieProvider (core)
* BasicAuth (core)
* OAuth (contrib)

**Comment by queenvictoria** [comment]
So how do you hand off from LDAP to session or cookie so that you don't authenticate on every page load?

`larowlan: I think it might be worth talking to znerol who did the session stuff in d8 or maybe neclimdul`

**Comment by queenvictoria** [comment]
ivanjaros left the chat room. (Client Quit)
[1:32pm] larowlan: queenvictoria: so AuthenticationProviderInterface looks like it is about detecting if the incoming request is for a logged in user
[1:33pm] tsvenson joined the chat room.
[1:33pm] larowlan: queenvictoria: which atm is basic auth and cookie
[1:33pm] larowlan: queenvictoria: but could also be digest or something i guess
[1:33pm] larowlan: queenvictoria: so ldap would still be cookie I reckon, but it would just need to handle login differently?
[1:34pm] queenvictoria: queenvictoria: yeah which is how i have it working at the moment-ish but not as a provider. once authentication is done it just sets the uid on the login form same as D7
[1:34pm] pwolanin joined the chat room.
[1:35pm] queenvictoria: larowlan: which allows the rest of the login process to flow
[1:35pm] jmolivas: heddn: we just got an issue about that topic, we are open to discuss it https://github.com/hechoendrupal/DrupalConsole/issues/1408
[1:35pm] larowlan: right

**Comment by queenvictoria** [comment]
Closing in favour of #11
